### DISCUSSION TITLE
Open/Save file dialog for Chrome

### DISCUSSION THREAD

**Discussion by neonvoidx** [asmvik/yabai#1179] [Q&A]
I have the following rule for Chrome:
`yabai -m rule --add app=".*Chrome.*" title=".*" opacity=1.0 space=top grid=1:5:2:0:3:1 manage=off`
but when I open a open/save file dialog in Chrome it is very odd appearing. Is there any way to fix this? I did a query and found that Google Chrome opens a `role="AXSheet"` for the dialog, so I tried adding:
`yabai -m rule --add app=".*Chrome.*" title="^$" role=".*AXSheet.*" opacity=1.0 manage=off` but doesn't seem to help.
Any suggestions? I just want my open/save dialog to be normal, or float in middle of screen even idc, but right now it expand the size of chrome's window, then when I close it, it repositions my chrome browser into an odd space

When I open a dialog:
![image](https://user-images.githubusercontent.com/25580051/157067644-18f1a33d-76bd-44e8-8c68-52893479358f.png)

After I close the dialog: (It repositions the original chrome window)
![image](https://user-images.githubusercontent.com/25580051/157068112-18e97ea0-b155-41f3-884c-f801ad686602.png)

**Comment by kkharji** [comment]
hmm I suggest getting the popup name. I don't use chrome but for safari I have this

```bash
yabai -m rule --add app="^Safari$" title = "^(General|(Tab|Password|Website|Extension)s|AutoFill|Se(arch|curity)|Privacy|Advance|Info|Settings|Copy)$"  manage ="off" layer = "above"
```

**Reply by kkharji** [reply]
to center them add `grid="6:4:1:1:2:4"`

**Reply by neonvoidx** [reply]
> hmm I suggest getting the popup name. I don't use chrome but for safari I have this
> 
> ```shell
> yabai -m rule --add app="^Safari$" title = "^(General|(Tab|Password|Website|Extension)s|AutoFill|Se(arch|curity)|Privacy|Advance|Info|Settings|Copy)$"  manage ="off" layer = "above"
> ```

I will try, but the title for these dialogs in chrome seems to be `title=""`

**Reply by kkharji** [reply]
🤔 try `app="^Google Chrome$" role="^AXSheet" manage=off grid="6:4:1:1:2:4"` also are u sure AXSheet isn't subrole?

Also I the first screenshoot seems to be finder window.  try
`app ="^Finder$"  title ="(Co(py|nnect)|Move|Info|Pref)" manage =off layer ="above" `

**Reply by neonvoidx** [reply]
> 🤔 try `app="^Google Chrome$" role="^AXSheet" manage=off grid="6:4:1:1:2:4"` also are u sure AXSheet isn't subrole?
> 
> Also I the first screenshoot seems to be finder window. try `app ="^Finder$" title ="(Co(py|nnect)|Move|Info|Pref)" manage =off layer ="above" `
`app ="^Finder$" title ="(Co(py|nnect)|Move|Info|Pref)" manage =off layer ="above"` doesn't work, also I did a diff using `yabai -m query --windows` and the only thing I changed was opening vs not having open a "Open" dialog from Chrome and the only diff was:
![image](https://user-images.githubusercontent.com/25580051/157085012-7f2a7406-c270-4e46-bc73-689be128dbf2.png)
so I assume this AXSheet is indeed it and belongs to chrome

**Reply by neonvoidx** [reply]
I tried with role as well, even though the output of query shows subrole, although doesn't look like `role` is considered a key
`yabai -m rule --add app=".*Chrome.*" role="AXSheet" manage=off grid="6:4:1:1:2:4" layer="above"
unknown key 'role'`

**Reply by kkharji** [reply]
Odd .. Well time to switch browsers 😆 . < seriously. 

🤔 I guess an issue should be opened in this case, add role and subrole to rules query?

**Reply by neonvoidx** [reply]
ya agree, and no I cant swap haha, I'm a UI dev and need Chrome since we target this mostly

**Reply by 24unix** [reply]
All except Safari and Fx use Blink anyway …

**Reply by kkharji** [reply]
Oh I see,  good luck, here my config https://gist.github.com/tami5/759b19871bd58ff20b47eeb1bdba50f1 if want extra rules other than chrome 😆 ofc

**Reply by neonvoidx** [reply]
https://github.com/koekeishiya/yabai/issues/1181 opened
